codegen/ace/src/comparison.rs

```diff
@@ -0,0 +1,64 @@
+use air_ir::{Operation, TraceAccess, Value};
+use std::cmp::Ordering;
+
+/// order by segment, offset, then column
+fn compare_access(a: &TraceAccess, b: &TraceAccess) -> Ordering {
+    let segment = a.segment.cmp(&b.segment);
+    if segment == Ordering::Equal {
+        let row = a.row_offset.cmp(&b.row_offset);
+        if row == Ordering::Equal {
+            a.column.cmp(&b.column)
+        } else {
+            row
+        }
+    } else {
+        segment
+    }
+}
+
+fn value_order(a: &Value) -> usize {
```

Comment:
This and the methods below look like an implementation of [`PartialOrd`](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html) which is like `Ord` but instead of `cmp(..) -> Ordering` we have `partial_cmp(..) -> Option<Ordering>`. Returning an `Option<usize>` here instead of `unreachable` would shift the responsibility to the caller to `unwrap` if they can ensure the ordering is always valid.